### PR TITLE
[5/N] Apply clang-tidy to aten/src/ATen/core 

### PR DESCRIPTION
**Pull request by cyyever**
Enlarge clang-tidy coverage to aten/src/ATen/core/* files

### CODE REVIEW

**Review by Skylion007 — COMMENTED** [review]
Some nits and some concerns about pb.h

File: aten/src/ATen/core/PhiloxRNGEngine.h [hunk 1]
```diff
@@ -113,6 +114,7 @@ class philox_engine {
       output_ = rand(counter, key, n_rounds);
       incr();
     }
+    // NOLINTNEXTLINE(*-narrowing-conversions)
```
[the remark this answers is not in the document]
Comment (cyyever) [reply aten/src/ATen/core/PhiloxRNGEngine.h:117]: In fact, I can't figure out where it is included. Greping aten/src/core returns no match.

File: .lintrunner.toml [hunk 2]
```diff
@@ -260,9 +261,12 @@ exclude_patterns = [
     # caffe2_pb.h, otherwise we'd have to build protos as part of this CI job.
```
[the remark this answers is not in the document]
Comment (cyyever) [reply .lintrunner.toml:261]: In fact, I can't figure out where it is included. Greping aten/src/core returns no match.

File: aten/src/ATen/core/PhiloxRNGEngine.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (cyyever) [reply aten/src/ATen/core/PhiloxRNGEngine.h:117]: at::detail::Array uses int as index type, should be changed to size_t

File: .lintrunner.toml [hunk 3]
```diff
@@ -254,17 +254,20 @@ command = [
...
 exclude_patterns = [
     # The negative filters below are to exclude files that include onnx_pb.h or
     # caffe2_pb.h, otherwise we'd have to build protos as part of this CI job.
     # CUDA files are also excluded.
+    '**/build/**',
     '**/fb/**',
     '**/*pb.h',
     '**/*CUDA*',
     '**/cuda/*pp',
+    'aten/src/ATen/core/*test.cpp',
```
Comment (Skylion007) [inline .lintrunner.toml:270]: Is this glob a little overbroad? Do all the test files use protobuff? Or include a header file that does?

Comment (cyyever) [reply .lintrunner.toml:270]: I want to exclude 2 tests, the one using protobuf and another with std::move

**Review by Skylion007 — APPROVED** [review]
Looks good!

File: aten/src/ATen/core/TensorAccessor.h [hunk 4]
```diff
@@ -156,7 +158,9 @@ class GenericPackedTensorAccessorBase {
   }
 protected:
   PtrType data_;
+  // NOLINTNEXTLINE(*c-arrays*)
```
Comment (Skylion007) [inline aten/src/ATen/core/TensorAccessor.h:161]: Actually why don't we use C++ std::array here? Do we need them in some constexpr context? Might be worth updating them.

**Review by Skylion007 — COMMENTED** [review]
Nit

File: aten/src/ATen/core/TensorAccessor.h (hunk 4 above)
Comment (cyyever) [reply aten/src/ATen/core/TensorAccessor.h:161]: I tired, but the constructor depends on the ability to decay C-array into pointers so the related class hierarchy and other places need to be changed too. Too many specific changes mean they should be put in a later PR.

**Pull request closed without merging**